File: src/main/java/easy/RemoveArrayDuplicates.java
package easy;

import java.util.ArrayList;
import java.util.Objects;

public class RemoveArrayDuplicates {
  public int removeDuplicates(ArrayList<Integer> a) {
    int count = 0;
    if (a.size() < 2) return a.size();
    for (Integer i : a) {
      if (!i.equals(a.get(count))) {
        count++;
        a.set(count, i);
      }
    }
    return count + 1;
  }

  public int solve(int[] nums) {
    int count = 0;
    if (nums.length < 2) return nums.length;
    for (int n : nums) {
      if (!Objects.equals(n, nums[count])) {
        count++;
        nums[count] = n;
      }
    }
    return count + 1;
  }
}

File: lib/easy/remove_array_duplicates.dart
int removeArrayDuplicates(List<int> nums) {
  if (nums.length <= 1) return 0;
  int count = 0;

  for (final val in nums) {
    if (val != nums[count]) {
      count++;
      nums[count] = val;
    }
  }
  return count + 1;
}
